### PR TITLE
refactor: post page to track referral

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- It seems as though we only have the post page that needs the hook for referral since we don't share links pointing to other places.
- Keeping this as draft until confirmed by the changes we will have on copy buttons.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-189 #done

### CODE REVIEW

**Comment by sshanzel** [comment]
Seems like it is the only place that needs the hook as discussed with Milan. cc @dailydotdev/web-team

**Review by nensidosari — APPROVED** [review]
Looks good

File: packages/shared/src/hooks/referral/useJoinReferral.ts [hunk 1]
```diff
@@ -47,6 +47,8 @@ export const useJoinReferral = (): void => {
         }
       }
     }
+
+    return null;
```
Comment (capJavert) [inline packages/shared/src/hooks/referral/useJoinReferral.ts:51]: Why is this return `null` needed?

Comment (sshanzel) [reply packages/shared/src/hooks/referral/useJoinReferral.ts:51]: Getting an error for some reason. I think `useQuery` needs to have a return value always, probably running til the end of time if not.

https://app.circleci.com/jobs/github/dailydotdev/apps/27458

**Comment by sshanzel** [comment]
> Looks good, but we also can also share user profile links, squad links, etc. Not sure if those pages don't need the same treatment?

How do we share user profile links? Do we have a button for it?

> squad links

What kind of Squad links? If invitation, that should be taken into account already.

**Pull request merged**